src/BuildScriptGenerator/DotNetCore/ProjectFileProviders/ProbeAndFindProjectFileProvider.cs

```diff
@@ -63,26 +69,82 @@ public string GetRelativePathToProjectFile(RepositoryContext context)
-                if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
+                if (ProjectFileHelpers.IsAzureBlazorWebAssemblyProject(sourceRepo, file))
                 {
-                    webAppProjects.Add(file);
+                    azureBlazorWasmProjects.Add(file);
+                    blazorWasmProjectExists = true;
                 }
                 else if (ProjectFileHelpers.IsAzureFunctionsProject(sourceRepo, file))
                 {
                     azureFunctionsProjects.Add(file);
+                    functionProjectExists = true;
+                }
+                else if (ProjectFileHelpers.IsAspNetCoreWebApplicationProject(sourceRepo, file))
+                {
+                    webAppProjects.Add(file);
+                    webAppProjectExists = true;
                 }
             }
 
-            projectFile = GetProject(webAppProjects);
-            if (projectFile == null)
+            // Assumption: some build option "--apptype" will be passed to oryx
+            // which will indicate if the app is an azure function type or static type app
+
+            // If there are multiple projects, we will look for --appty to detect corresponding project.
+            // for example azurefunction and blazor both projects can reside
+            // at the same repo, so more than 2 csprojs will be found. Now we will
+            // look for --apptype value to determine which project needs to be built
+            string oryxAppTypeEnv = string.Empty;
+
+            if (_options != null
+                && _options.Value != null
+                && _options.Value.OryxAppType != null)
             {
-                projectFile = GetProject(azureFunctionsProjects);
+                oryxAppTypeEnv = _options.Value.OryxAppType;
+                _logger.LogInformation($"ProbeAndProjectFileProvider: {Constants.OryxAppType} is set to {oryxAppTypeEnv}");
+
+                if (functionProjectExists && oryxAppTypeEnv.ToLower().Contains("functions"))
+                {
+                    projectFile = GetProject(azureFunctionsProjects);
+                }
+                else if (blazorWasmProjectExists
+                    && (oryxAppTypeEnv.ToLower().Contains("static-sites")
+                    || oryxAppTypeEnv.ToLower().Contains("blazor-wasm")))
+                {
+                    projectFile = GetProject(azureBlazorWasmProjects);
+                }
+                else
+                {
+                    _logger.LogDebug($"Invalid value {oryxAppTypeEnv} for env:{Constants.OryxAppType}. Currently, supported values are 'functions', 'blazor-wasm', 'static-sites'");
+                }
+            }
+            else
+            {
+                // If multiple project exists, webapp gets priority if appType is not set
+                // orderwise webapp then blazor-wasm and then functions
+                if (projectFile == null && webAppProjectExists)
```

Comment:
I would change this logic like following:
1. Merge webAppProjects + azureBlazorWasmProjects + azureFunctionsProjects into one list
2. GetProject(merged list), that will throw ambiguity exception if we detected multiple projects

Reply:
imo its better to default to webapp when no value is provided to apptype. but let's discuss it